atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,690 @@
+
+        self._nomad_to_particles_group_map = {
+            'positions': 'position',
+            'velocities': 'velocity',
+            'forces': 'force',
+            'labels': 'species_label',
+            'label': 'force_field_label',
+            'mass': 'mass',
+            'charge': 'charge',
+        }
+
+        self._nomad_to_box_group_map = {
+            'lattice_vectors': 'edges',
+            'periodic': 'boundary',
+            'dimension': 'dimension'
+        }
+
+    def format_times(self, times):
+        return np.around(times.magnitude * ureg.convert(1.0, times.units, self._time_unit), 5)
+
+    def parse_atom_parameters(self):
+        if self._n_atoms is None:
+            return {}
+        self._atom_parameters = {}
+        n_atoms = self._n_atoms[0]  # TODO Extend to non-static n_atoms
+
+        atom_parameter_keys = ['label', 'mass', 'charge']
+        for key in atom_parameter_keys:
+            value = self._data_parser.get(f'{self._path_group_particles_all}.{self._nomad_to_particles_group_map[key]}')
+            if value is not None:
+                self._atom_parameters[key] = value
+            else:
+                continue
+            if isinstance(self._atom_parameters[key], h5py.Group):
+                self.logger.warning('Time-dependent atom parameters currently not supported. Atom parameter values will not be stored.')
+                continue
+            elif len(self._atom_parameters[key]) != n_atoms:
+                self.logger.warning('Inconsistent length of some atom parameters. Atom parameter values will not be stored.')
+                continue
+
+    def parse_system_info(self):
+        self._system_info = {'system': {}, 'calculation': {}}
+        particles_group = self._data_parser.get(self._path_group_particles_all)
+        positions = self._data_parser.get(self._path_value_positions_all)
+        n_frames = self._n_frames
+        if particles_group is None or positions is None or positions is None: # For now we require that positions are present in the H5MD file to store other particle attributes
+            self.logger.warning('No positions available in H5MD file. Other particle attributes will not be stored')
+            return self._system_info
+
+        def get_value(value, steps, path=''):
+            if value is None:
+                return value
+            if isinstance(value, h5py.Group):
+                path_value = f'{path}.value' if path else 'value'
+                value = self._data_parser.get(path_value)
+                path_step = f'{path}.step' if path else 'step'
+                attr_steps = self._data_parser.get(path_step)
+                if value is None or attr_steps is None:
+                    self.logger.warning('Missing values or steps in particle attributes. These attributes will not be stored.')
+                    return None
+                elif attr_steps.sort() != steps.sort() if attr_steps is not None else False:
```

Comment:
.sort() will return None, do you want to compare the sorted list? if so, this would not work and will always return true.

Reply:
Thank you! I did not realize this. `elif sorted(attr_steps) != sorted(steps) if attr_steps is not None else False:` works right?

Actually, I think I only need `elif sorted(attr_steps) != sorted(steps)` since I already check `attr_steps is None` in the first if statement.

Reply:
> Thank you! I did not realize this. `elif sorted(attr_steps) != sorted(steps) if attr_steps is not None else False:` works right?
> 
> Actually, I think I only need `elif sorted(attr_steps) != sorted(steps)` since I already check `attr_steps is None` in the first if statement.

o yes you are right.